db/model/Variable.ts

```diff
@@ -280,23 +285,110 @@ export async function updateAllChartsThatInheritFromIndicator(
                     cc.full = ?,
                     cc.updatedAt = ?,
                     c.updatedAt = ?
-                WHERE c.id = ?
+                WHERE cc.id = ?
             `,
-            [JSON.stringify(fullConfig), updatedAt, updatedAt, chart.chartId]
+            [
+                JSON.stringify(fullConfig),
+                updatedAt,
+                updatedAt,
+                chart.chartConfigId,
+            ]
         )
     }
 
     // let the caller know if any charts were updated
     return inheritingCharts
 }
 
+async function findAllMultiDimViewsThatInheritFromIndicator(
+    trx: db.KnexReadonlyTransaction,
+    variableId: number
+): Promise<
+    {
+        chartConfigId: string
+        patchConfig: GrapherInterface
+        isPublished: boolean
+    }[]
+> {
+    const charts = await db.knexRaw<{
+        chartConfigId: DbPlainMultiDimXChartConfig["chartConfigId"]
+        patchConfig: DbRawChartConfig["patch"]
+        isPublished: boolean
+    }>(
+        trx,
+        `-- sql
+            SELECT
+                mdxcc.chartConfigId as chartConfigId,
+                cc.patch as patchConfig,
+                md.published as isPublished
+            FROM multi_dim_data_pages md
+                JOIN multi_dim_x_chart_configs mdxcc ON mdxcc.multiDimId = md.id
+                JOIN chart_configs cc ON cc.id = mdxcc.chartConfigId
+            WHERE mdxcc.variableId = ?
+        `,
+        [variableId]
+    )
+    return charts.map((chart) => ({
+        ...chart,
+        patchConfig: parseChartConfig(chart.patchConfig),
+    }))
+}
+
+export async function updateAllMultiDimViewsThatInheritFromIndicator(
+    trx: db.KnexReadWriteTransaction,
+    variableId: number,
+    {
+        updatedAt,
+        patchConfigETL,
+        patchConfigAdmin,
+    }: {
+        updatedAt: Date
+        patchConfigETL?: GrapherInterface
+        patchConfigAdmin?: GrapherInterface
+    }
+): Promise<
+    {
+        chartConfigId: string
+        patchConfig: GrapherInterface
+        isPublished: boolean
+    }[]
+> {
+    const inheritingViews = await findAllMultiDimViewsThatInheritFromIndicator(
+        trx,
+        variableId
+    )
+
+    for (const view of inheritingViews) {
+        const fullConfig = mergeGrapherConfigs(
+            patchConfigETL ?? {},
+            patchConfigAdmin ?? {},
+            view.patchConfig
+        )
+        await db.knexRaw(
+            trx,
+            `-- sql
+                UPDATE chart_configs
+                SET
+                    full = ?,
+                    updatedAt = ?
```

Comment:
UpdatedAt should auto-update - I would remove the explicit update from here unless there is a good reason to do it manually?

Reply:
I copied and tweaked existing code, and this is what we did there already. Checked just now and `updatedAt` is not set to auto-update, and I'm not sure why. @sophiamersmann do you remember why we set it manually?

Reply:
I initially switched to auto-update and then switched back to manual updates when Mojmir noticed that the updatedAt of the chart and the updatedAt of the config were out of sync for some charts. If it’s automatic updates then that can happen when you run a migration on one of the tables, for example. I don’t recall why Mojmir needed these to be the same, chart-diff maybe?

Reply:
Ah yeah, I vaguely remember that. Thanks!

Reply:
Ok cool, if there is a reason that that is fine. Could you just add a comment (can just quote the vague statement of Sophia even :) ) so that we don't accidentally delete this in the future?

Reply:
Good point, added comments.